src/glview/cgal/CGALRenderer.cc

```diff
@@ -333,3 +333,81 @@ BoundingBox CGALRenderer::getBoundingBox() const
   }
   return bbox;
 }
+double calculateLinePointDistance(const Vector3d &l1, const Vector3d &l2, const Vector3d &pt, double & dist_lat) {
+    Vector3d d = (l2 - l1).normalized();
+    dist_lat = (pt-l1).dot(d);
+    return (l1 + d * dist_lat-pt).norm();
+}
+
+double calculateLineLineDistance(const Vector3d &l1b, const Vector3d &l1e, const Vector3d &l2b, const Vector3d &l2e, double &dist_lat)
+{
+	double d;
+	Vector3d v1=l1e-l1b;
+	Vector3d v2=l2e-l2b;
+	Vector3d n=v1.cross(v2);
+	if(n.norm() == 0) {
+		return calculateLinePointDistance(l1b,l1e,l2b,d);
+	}
+	double t=n.norm();
+	n.normalize();
+    	d=n.dot(l1b-l2b);
+	dist_lat=(v2.cross(n)).dot(l2b-l1b)/t;
+	return d;
+}
+
+std::vector<SelectedObject> CGALRenderer::findModelObject(Vector3d near_pt, Vector3d far_pt,int mouse_x, int mouse_y, double tolerance) {
+  std::vector<SelectedObject> results;
+  double dist_near;
+  double dist_nearest=NAN;
+  Vector3d pt1_nearest;
+  Vector3d pt2_nearest;
+  for (const auto& p : this->getPolyhedrons()) {
+  }
```

Comment:
?

Reply:
is there any data there ?

Reply:
I am happy to process the vertices coming from CGAL, however,  I was not able to find an usable template code for that. 
we can improve here any time.